tron/bin/check_tron_datastore_staleness.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python3.6
+import argparse
+import logging
+import os
+import sys
+import time
+
+import pytz
+
+from tron.config import manager
+from tron.config import schema
+from tron.serialize.runstate.dynamodb_state_store import DynamoDBStateStore
+
+# Default values for arguments
+DEFAULT_WORKING_DIR = '/var/lib/tron/'
+DEFAULT_CONF_PATH = 'config/'
+DEFAULT_STALENESS_THRESHOLD = 30
+log = logging.getLogger('tron_stateless_alert')
+
+
+def get_last_run_time(job):
+    '''
+    Get all sorted timestamps, and only count the actions that actually ran
+    '''
+    timestamps = []
+    job_runs = job['runs']
+    for run in job_runs:
+        for action in run['runs']:
+            if action.get('start_time') and action.get('state') != 'scheduled':
+                timestamps.append(action.get('start_time'))
+    timestamps.sort()
+    return timestamps[-1] if timestamps else None
+
+
+def parse_cli():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-w",
+        "--working-dir",
+        default=DEFAULT_WORKING_DIR,
+        help="Working directory for the Tron daemon, default %(default)s",
+    )
+    parser.add_argument(
+        "-c",
+        "--config-path",
+        default=DEFAULT_CONF_PATH,
+        help="File path to the Tron configuration file",
+    )
+    parser.add_argument(
+        "--job-for-stateless-alert",
+        required=True,
+        help="The job name to read timestamp from",
+    )
+    parser.add_argument(
+        "--staleness-threshold",
+        default=DEFAULT_STALENESS_THRESHOLD,
+        help="how long (in seconds) to wait to alert after the last timestamp of the job",
+    )
+    args = parser.parse_args()
+    args.working_dir = os.path.abspath(args.working_dir)
+    args.config_path = os.path.join(
+        args.working_dir,
+        args.config_path,
+    )
+    return args
+
+
+def read_config(args):
+    return manager.ConfigManager(args.config_path).load().get_master().state_persistence
+
+
+def main():
+    # Fetch configs. You can find the arguments in puppet.
+    args = parse_cli()
+    persistence_config = read_config(args)
+    store_type = schema.StatePersistenceTypes(persistence_config.store_type)
+    job_for_stateless_alert = args.job_for_stateless_alert
+
+    # Alert for DynamoDB
+    if store_type == schema.StatePersistenceTypes.dynamodb:
+        # Fetch job state from dynamodb
+        dynamodb_region = persistence_config.dynamodb_region
+        table_name = persistence_config.table_name
+        store = DynamoDBStateStore(table_name, dynamodb_region)
+        key = store.build_key('job_state', job_for_stateless_alert)
+        try:
+            job = store.restore([key])[key]
+        except Exception as e:
+            logging.error(f'Failed to retreive status for job {job_for_stateless_alert} due to {e}')
```

Comment:
It's usually better to print exceptions with repr, e.g `{e!r}`, because that provides more information. For example, for `e = KeyError('a_key')`, str(e) is `'a_key'` but repr(e) is `KeyError('a_key')`

You could also just use `logging.exception` to show the traceback.